bluefin-random/src/Bluefin/Random.hs

```diff
@@ -0,0 +1,45 @@
+module Bluefin.Random (
+  Random,
+  runRandom,
+  withRandom
+  ) where
+
+import Prelude ((.), pure, (<$>), ($), flip)
+
+import qualified System.Random as Rnd
+import Bluefin.State (State, evalState, get, put)
+import Bluefin.Eff (Eff, (:&))
+import qualified System.Random.Stateful as Rnd
+import Data.Bifunctor (second)
+
+newtype Random g e = Random (State g e)
+
+newtype RandomPure g = RandomPure { unRandomPure :: g }
+  deriving newtype (Rnd.RandomGen)
+
+instance Rnd.RandomGen g => Rnd.StatefulGen (Random g e) (Eff (e :& es)) where
+  uniformWord64 = flip withRandom Rnd.genWord64
+
+  uniformByteArrayM pinned size = flip withRandom
+    $ Rnd.uniformByteArray pinned size
+
+instance Rnd.RandomGen g => Rnd.FrozenGen (RandomPure g) (Eff (e :& es)) where
```

Comment:
```suggestion
instance Rnd.RandomGen g => Rnd.FrozenGen (RandomPure g) (Eff es) where
```

Reply:
Hmm, actually this doesn't seem right. I'll have to think about the best way of doing this.

Reply:
Yes I missed it last night that this is fixing the order of effects. It may be the case that I can't create an associated type `MutableGen` that fits. I will have a go at another approach later.

Reply:
Actually I can just add a tag to random pure!
```haskell
instance (Rnd.RandomGen g, e :> es) => Rnd.FrozenGen (RandomPure g e) (Eff es) where
  type MutableGen (RandomPure g e) (Eff es) = Random g e
```

Reply:
This leads to pretty painful usage in the example: `n <- randomRM @Int (1, 5) (mapHandle r)`

Reply:
Ah yes, I should make sure I'm always reading the up-to-date docs. OK, then I agree that adding the tag to `RandomPure` is the way to go.